File: internal/model/pet.go
package model

type Pet struct {
	BaseModel
	OrgID        uint   `json:"org_id" gorm:"index"`
	Name         string `json:"name" gorm:"size:50"`
	Species      string `json:"species" gorm:"size:20;index"`
	Breed        string `json:"breed" gorm:"size:50"`
	Age          int    `json:"age"`
	Gender       string `json:"gender" gorm:"size:10"`
	HealthStatus string `json:"health_status" gorm:"size:50"`
	Description  string `json:"description" gorm:"type:text"`
	CoverImage   string `json:"cover_image" gorm:"size:500"`
	Status       string `json:"status" gorm:"size:20;default:available;index"`

	Organization Organization `json:"organization" gorm:"foreignKey:OrgID"`
	Images       []PetImage   `json:"images" gorm:"foreignKey:PetID"`
}

type PetImage struct {
	BaseModel
	PetID     uint   `json:"pet_id" gorm:"index"`
	ImageURL  string `json:"image_url" gorm:"size:500"`
	SortOrder int    `json:"sort_order" gorm:"default:0"`
}

File: web/src/api/pet.ts
import request from "./request";



export function getPets(params:{
    page?:number;
    pageSize?:number;
    species?:string;
    gender?:string;
    status?:string;
    org_id?:number;
}) {
    return request.get('/pets', {params});
}

export function getPetDetail(id: number) {
    return request.get(`/pets/${id}`);
}

export function createPet(data:{
    name: string;
    species?:string;
    breed?: string;
    age: number;
    gender: string;
    description:string;
    cover_image?:string;
}) {
    return request.post('/pets', data);
}

// 删除宠物
export function deletePet(id: number) {
    return request.delete(`/pets/${id}`);
}

// 更新宠物
export function updatePet(id: number, data: {
    name?: string;
    species?: string;
    breed?: string;
    age?: number;
    gender?: string;
    description?: string;
    cover_image?: string;
    status?: string;
}) {
    return request.put(`/pets/${id}`, data);
}

// 上传宠物图片（修正路径）
export function uploadPetImage(petId: number, file: File, sortOrder?: number) {
    const formData = new FormData();
    formData.append('image', file);
    if (sortOrder !== undefined) {
        formData.append('sort_order', String(sortOrder));
    }
    return request.post(`/pets/${petId}/images`, formData);
}